pipestat/backend.py

```diff
@@ -0,0 +1,1226 @@
+import sys
+import os
+from abc import ABC
+from glob import glob
+from logging import getLogger
+from yacman import YAMLConfigManager
+from ubiquerg import create_lock, remove_lock, expandpath
+from contextlib import contextmanager
+
+from sqlalchemy import text
+from sqlmodel import Session, select as sql_select
+
+from .const import *
+from .exceptions import *
+from .helpers import *
+
+if int(sys.version.split(".")[1]) < 9:
+    from typing import List, Dict, Any, Optional, Union
+else:
+    from typing import *
+
+_LOGGER = getLogger(PKG_NAME)
+
+
+class PipestatBackend(ABC):
```

Comment:
I would suggest separate 3 classes into 3 different files. This approach would enhance code readability and maintainability.

Reply:
I think I agree with this. It might make sense to put these into a `backends/` subfolder, like:

`backends/abstract.py`
`backends/file_backend.py`
`backends/sql_backend.py`

or something like that. I don't know if we have to do that right now though. Could continue to think about this.

Reply:
I moved these to their separate files as I agree this is a good idea.